src/clincoded/static/components/add_external_resource.js

```diff
@@ -785,29 +811,43 @@ function carQueryResource() {
                             // Remove the 'tempAlleles' object at this point regardless
                             // whether the preceding evaluations are executed
                             if (data['tempAlleles']) delete data['tempAlleles'];
-                            this.setState({queryResourceBusy: false, tempResource: data, resourceFetched: true});
                         } else {
                             // Fall back to CAR data without the canonical transcript title if there is no ensembl transcript
                             if (data['tempAlleles']) delete data['tempAlleles'];
-                            this.setState({queryResourceBusy: false, tempResource: data, resourceFetched: true});
                         }
-                    }).catch(err => {
+                    }
+                    catch (err) {
                         // Error in VEP get request
                         console.warn('Error in querying Ensembl VEP data = %o', err);
                         // Fall back to CAR data
                         if (data['tempAlleles']) delete data['tempAlleles'];
-                        this.setState({queryResourceBusy: false, tempResource: data, resourceFetched: true});
-                    });
+                    }
                 } else {
                     // Fall back to CAR data without the canonical transcript title if parsing fails
                     if (data['tempAlleles']) delete data['tempAlleles'];
-                    this.setState({queryResourceBusy: false, tempResource: data, resourceFetched: true});
                 }
             } else {
                 // in case the above two fail (theoretically a 404 json response, but an error is thrown instead (see below))
                 this.setFormErrors('resourceId', 'CA ID not found');
-                this.setState({queryResourceBusy: false, resourceFetched: false});
+                finalState.resourceFetched = false;
```

Comment:
To maintain previous `setState` logic, I'd recommend adding a removal of `tempResource` (from `finalState`) here.